x/incentives/keeper/distribute.go

```diff
@@ -602,6 +602,27 @@ func (k Keeper) distributeInternal(
 		// Get distribution epoch duration. This is used to calculate the emission rate.
 		currentEpoch := k.GetEpochInfo(ctx)
 
+		// Validate that the gauge's corresponding uptime is authorized.
+		authorizedUptimes := k.clk.GetParams(ctx).AuthorizedUptimes
```

Comment:
I think we should pass authorized uptimes as a func input param instead of retrieving it for every single gauge, this could be expensive.

Reply:
Hmm I agree that this might be expensive, but unfortunately `distributeInternal` is used for all gauges, not just CL gauges, so authorized uptimes as a func input would be super odd. Happy to continue discussion offline

Reply:
This is only in the noLock gauge (not superfluid) and can be altered state compatibly so i'm for filing this into a GH issue and doing later!